lib/templateTagFactory.js

```diff
@@ -129,6 +129,7 @@ class TemplateTagFactory extends BaseFactory {
         const nameObj = parseTemplateConfigName(config);
         const result = hoek.applyToDefaults(config, nameObj);
 
+        result.templateType = this.getTemplateType();
```

Comment:
Wouldn't this throw an error?
Whey did we remove `getTemplateType()` and `_getTemplateType()` methods?
https://github.com/screwdriver-cd/models/pull/585/files#diff-664ca85ec42eff02c175a8c344671c9c6d55f78fa19bd34a2e37ef238e20f225R185-R193

Reply:
I think they were duplicates

Reply:
`TemplateTagFactory` is the base class which will be extended by two concrete classes `JobTemplateTagFactory` and `TemplateTagFactory`.

In the base class we need to have `getTemplateType()` will should actually throw an error saying "unimplemented". And the concrete classes should override this to return a valid type.

Since the base class is already being used to API for Job Template workflows, the method should to temporarily return 'JOB' until we refactor the API to use `JobTemplateTagFactory`.

That said, I don't have a strong option on whether we need to have both `getTemplateType()` and `_getTemplateType()` or  just consolidate with one method.
Two methods were added to follow the convention being followed in other classes.

Reply:
The functions are there https://github.com/screwdriver-cd/models/blob/1479e0381137b67d55738247dddda0ddc5c6cd17/lib/templateTagFactory.js#L95-L99
The deleted ones were duplicates

Reply:
Ahh ... completely forgot that we had already merged some code earlier https://github.com/screwdriver-cd/models/pull/586/files